src/pykeen/training/training_loop.py

```diff
@@ -679,3 +810,85 @@ def _free_graph_and_cache(self):
         self.model.regularizer.reset()
         # The cache of the previous run has to be freed to allow accurate memory availability estimates
         torch.cuda.empty_cache()
+
+    def _save_state(self, path: Union[str, pathlib.Path], stopper: Optional[Stopper] = None) -> None:
+        """Save the state of the training loop.
+
+        :param path:
+            Path of the file where to store the state in.
+        :param stopper:
+            An instance of :class:`pykeen.stopper.EarlyStopper` with settings for checking
+            if training should stop early
+        """
+        logger.debug("=> Saving checkpoint.")
+
+        if stopper is None:
+            stopper_dict = dict()
+        else:
+            stopper_dict = stopper.get_summary_dict()
+
+        # Only if a cuda device is available, the random state is accessed
+        if torch.cuda.is_available():
+            torch_cuda_random_state = torch.cuda.get_rng_state()
+        else:
+            torch_cuda_random_state = None
+
+        torch.save(
+            {
+                'epoch': self._epoch,
+                'loss': self.losses_per_epochs,
+                'model_state_dict': self.model.state_dict(),
+                'optimizer_state_dict': self.optimizer.state_dict(),
+                'checksum': self.checksum,
+                'random_seed': self.model._random_seed,
```

Comment:
is this necessary for continuing training? Since the current state when it's saved is also here?

Reply:
in general we don't do a perfect job at storing all configuration settings, but I don't know if having the training loop take care of this makes so much sense

Reply:
> is this necessary for continuing training? Since the current state when it's saved is also here?

As discussed before, the random_seed is required for documentation purposes.

Reply:
> in general we don't do a perfect job at storing all configuration settings, but I don't know if having the training loop take care of this makes so much sense

If we want, we could take over all "training loop relevant" parts from the pipeline and log them in the training loop as well.